org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/ProjectManager.java

```diff
@@ -0,0 +1,216 @@
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *******************************************************************************/
+package org.eclipse.xtext.ide.server;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+import org.eclipse.emf.common.util.URI;
+import org.eclipse.emf.ecore.resource.Resource;
+import org.eclipse.xtext.build.BuildRequest;
+import org.eclipse.xtext.build.IncrementalBuilder;
+import org.eclipse.xtext.build.IndexState;
+import org.eclipse.xtext.diagnostics.Severity;
+import org.eclipse.xtext.resource.IExternalContentSupport;
+import org.eclipse.xtext.resource.IResourceDescription;
+import org.eclipse.xtext.resource.IResourceServiceProvider;
+import org.eclipse.xtext.resource.XtextResourceSet;
+import org.eclipse.xtext.resource.impl.ChunkedResourceDescriptions;
+import org.eclipse.xtext.resource.impl.ProjectDescription;
+import org.eclipse.xtext.resource.impl.ResourceDescriptionsData;
+import org.eclipse.xtext.util.CancelIndicator;
+import org.eclipse.xtext.util.IFileSystemScanner;
+import org.eclipse.xtext.validation.Issue;
+import org.eclipse.xtext.workspace.IProjectConfig;
+import org.eclipse.xtext.workspace.ISourceFolder;
+import org.eclipse.xtext.workspace.ProjectConfigAdapter;
+import org.eclipse.xtext.xbase.lib.Procedures.Procedure2;
+
+import com.google.common.collect.ImmutableList;
+import com.google.inject.Inject;
+import com.google.inject.Provider;
+
+/**
+ * @author Sven Efftinge - Initial contribution and API
+ * @since 2.11
+ */
+public class ProjectManager {
+	@Inject
+	protected IncrementalBuilder incrementalBuilder;
+
+	@Inject
+	protected Provider<XtextResourceSet> resourceSetProvider;
+
+	@Inject
+	protected IResourceServiceProvider.Registry languagesRegistry;
+
+	@Inject
+	protected IFileSystemScanner fileSystemScanner;
+
+	@Inject
+	protected IExternalContentSupport externalContentSupport;
+
+	private IndexState indexState = new IndexState();
+
+	private URI baseDir;
+
+	private Procedure2<? super URI, ? super Iterable<Issue>> issueAcceptor;
+
+	private Provider<Map<String, ResourceDescriptionsData>> indexProvider;
+
+	private IExternalContentSupport.IExternalContentProvider openedDocumentsContentProvider;
+
+	private XtextResourceSet resourceSet;
+
+	private ProjectDescription projectDescription;
+
+	private IProjectConfig projectConfig;
+
+	public void initialize(ProjectDescription description, IProjectConfig projectConfig,
+			Procedure2<? super URI, ? super Iterable<Issue>> acceptor,
+			IExternalContentSupport.IExternalContentProvider openedDocumentsContentProvider,
+			Provider<Map<String, ResourceDescriptionsData>> indexProvider, CancelIndicator cancelIndicator) {
+		projectDescription = description;
```

Comment:
sometimes fields prefixed `this.` sometimes not

Reply:
`this.` is used where necessary (eg. field and parameter have the same name). Will make this consistent here.

Reply:
Done